pkg/throttle/byterate.go

```diff
@@ -0,0 +1,48 @@
+package throttle
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+var unmarshalRegex = regexp.MustCompile(`^(\d+)\s?([KMGT]?Bps)$`)
+
+type Byterate int
+
+const (
+	Bps  Byterate = 1
+	KBps          = 1024 * Bps
+	MBps          = 1024 * KBps
+	GBps          = 1024 * MBps
+	TBps          = 1024 * GBps
+)
+
+func (br *Byterate) UnmarshalText(b []byte) error {
+	comps := unmarshalRegex.FindStringSubmatch(string(b))
+	if len(comps) != 3 {
+		return fmt.Errorf("invalid byterate format %s should be n Bps, n KBps, n MBps, n GBps, or n TBps", string(b))
+	}
+	v, err := strconv.Atoi(comps[1])
+	if err != nil {
+		return err
+	}
+	unitStr := comps[2]
+	var unit Byterate
+	switch unitStr {
+	case "Bps":
+		unit = Bps
+	case "KBps":
+		unit = KBps
+	case "MBps":
+		unit = MBps
+	case "GBps":
+		unit = GBps
+	case "TBps":
+		unit = TBps
```

Comment:
Maybe we should write something about these? The difference between KBps, kb/s, Kbps and kbps is always hard.

Reply:
The argument is documented both the command arg and in Helm values. The units follow general nomenclature that B = byte and b = bit. Allowing users to specify Mbps makes no sense as we are writing data in byte chunks.